libvast/src/chunk.cpp

```diff
@@ -187,24 +188,29 @@ caf::error inspect(caf::serializer& sink, const chunk_ptr& x) {
     });
 }
 
-caf::error inspect(caf::deserializer& source, chunk_ptr& x) {
+caf::error inspect(caf::deserializer&, chunk_ptr&) {
+  return caf::make_error(ec::parse_error, "caf::deserializer inspection "
+                                          "instead of legacy_deserializer");
+}
+
+bool inspect(detail::legacy_deserializer& source, chunk_ptr& x) {
   uint32_t size = 0;
-  if (auto err = source(size))
-    return err;
+  if (!source(size))
+    return false;
   if (size == 0) {
     x = nullptr;
-    return caf::none;
+    return true;
   }
   auto buffer = std::make_unique<chunk::value_type[]>(size);
   const auto data = buffer.get();
-  if (auto err = source.apply_raw(size, data)) {
+  if (!source.apply_raw(size, data)) {
     x = nullptr;
-    return caf::none;
+    return false; // TODO originally returned true
```

Comment:
That was definitely a bug.

Reply:
Resolved in commit c0311cfd71c165a71bc0e8c4f941fa938572e14d